distribution/push_v2.go

```diff
@@ -240,10 +240,10 @@ func (pd *v2PushDescriptor) DiffID() layer.DiffID {
 }
 
 func (pd *v2PushDescriptor) Upload(ctx context.Context, progressOutput progress.Output) (distribution.Descriptor, error) {
-	if fs, ok := pd.layer.(layer.ForeignSourcer); ok {
-		if d := fs.ForeignSource(); d != nil {
+	if fs, ok := pd.layer.(distribution.Describable); ok {
+		if d := fs.Descriptor(); len(d.URLs) > 0 {
```

Comment:
We should only use the mediatype for determining whether or not this is a foreign layer.

Although, that may be part of the other PR.

cc @runcom